psydac/api/ast/fem.py

```diff
@@ -474,8 +476,9 @@ def _create_ast_bilinear_form(terminal_expr, atomic_expr_field,
             for v in sub_tests+sub_trials:
                 stmts += construct_logical_expressions(v, nderiv)
 
-            if atomic_expr_field:
-                stmts += list(eval_field.inits)
```

Comment:
was the bug here?

Reply:
Yes, it was only printing the instructions for the last field.

Reply:
I see